libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java

```diff
@@ -39,8 +40,20 @@ public MotionSensor(Context appCtx) {
         // If no TYPE_SIGNIFICANT_MOTION is available, use alternate means to sense motion
         if (motionSensor == null) {
             motionSensor = new LegacyMotionSensor(mAppContext);
-            AppGlobals.guiLogInfo("Device has legacy motion sensor.");
         }
+        setTypeFromPrefs();
+    }
+
+    private void setTypeFromPrefs() {
+        if (Prefs.getInstance(mAppContext).getIsMotionSensorTypeSignificant()) {
+            motionSensor = SignificantMotionSensor.getSensor(mAppContext);
+        }
+    }
+
+    // Call when the scanning has stopped (not when paused).
+    // This class can use this event to update its state
+    public void scannerFullyStopped() {
+        setTypeFromPrefs();
```

Comment:
Oh. We have a second instance of calling `setTypeFromPrefs`.  Why are we calling this from `stopScanning()`?  Do we really want to be rebinding the signficant motion sensor when we stop scanning?